biogeochem/FatesSoilBGCFluxMod.F90

```diff
@@ -0,0 +1,1114 @@
+       if(n_uptake_mode.eq.coupled_n_uptake .or. p_uptake_mode.eq.coupled_p_uptake)then
+
+          nlev_eff_soil = max(bc_in(s)%max_rooting_depth_index_col, 1)
+          
+          if(fates_ncomp_scaling.eq.pft_ncomp_scaling) then
+             
+             allocate(rootfrac_pft_decomp(numpft,bc_in(s)%nlevdecomp))
+             allocate(fnrt_c_pft(numpft))
+             
+             ! Construct the pft x decomp layer fine-root profile
+             ! USe the profile associated with hydraulic uptake! 
+             
+             do pft = 1, numpft
+                rootfrac_pft_decomp(pft,:) = 0._r8
+                call set_root_fraction(sites(s)%rootfrac_scr, pft, sites(s)%zi_soil)
+                
+                do j = 1,nlev_eff_soil
+                   id = bc_in(s)%decomp_id(j)  ! Map from soil layer to decomp layer    
+                   rootfrac_pft_decomp(pft,id) = rootfrac_pft_decomp(pft,id) + & 
+                        sites(s)%rootfrac_scr(j)
+                end do
+                
+             ! Make sure that the rootfrac profile sums to unity
+                rootfrac_pft_decomp(pft,:) = rootfrac_pft_decomp(pft,:)/sum(rootfrac_pft_decomp(pft,:))
+             end do
+             
+             ! Calculate the total fineroot mass for each PFT, so we can weight
+             fnrt_c_pft(:) = 0._r8
+             cpatch => sites(s)%oldest_patch
+             do while (associated(cpatch))
+                ccohort => cpatch%tallest
+                do while (associated(ccohort))
+                   pft   = ccohort%pft
+                   fnrt_c_pft(pft) = fnrt_c_pft(pft) + &
+                        ccohort%prt%GetState(fnrt_organ, all_carbon_elements)*ccohort%n
+                   ccohort => ccohort%shorter
+                end do
+                cpatch => cpatch%younger
+             end do
+          end if  ! end if(fates_ncomp_scaling.eq.pft_ncomp_scaling) then
+          
+          ! --------------------------------------------------------------------------------
+          ! Now that we have the arrays ready for downscaling (if needed)
+          ! loop through all cohorts and acquire nutrient
+          ! --------------------------------------------------------------------------------
+
+          if(n_uptake_mode.eq.coupled_n_uptake) then
+             icomp = 0
+             cpatch => sites(s)%oldest_patch
+             do while (associated(cpatch))
+                ccohort => cpatch%tallest
+                do while (associated(ccohort))
+                   pft   = ccohort%pft
+                   if(fates_ncomp_scaling.eq.cohort_ncomp_scaling) then
+                      icomp = icomp+1
+
+                      ! N Uptake:  Convert g/m2/day -> kg/plant/day
+                      ccohort%daily_n_uptake = ccohort%daily_n_uptake + &
+                           sum(bc_in(s)%plant_n_uptake_flux(icomp,:))*kg_per_g*AREA/ccohort%n
+
+                   else
```

Comment:
what is the situation in this 'else' statement? 
Apologies for requesting all of this hand-holding comment statements....

Reply:
This if structure options the two ways of scaling the boundary conditions.  There are two options.  Either send all the cohorts as independent competitors to the land-models biogeochemistry competition scheme, or group all cohorts into pfts and have pfts act as competitors.  The second side of this if performs the latter.